### ISSUE TITLE
trailing_closure rule with NSAnimationContext runAnimationGroup(_:completionHandler:)

### ISSUE THREAD

**Issue by Eitot** [realm/SwiftLint#1754]
The rule is triggered when a method has two closure parameters and the second one is optional, such as [`runAnimationGroup(_:completionHandler:)`](https://developer.apple.com/documentation/appkit/nsanimationcontext/1529847-runanimationgroup). Trailing-closure syntax is not available in that case (I believe).

Examples:
```swift
// This does not work
NSAnimationContext.runAnimationGroup({ context in
    context.duration = 0.4
    backgroundView.animator().alphaValue = isShown ? 1 : 0
})

// This does
NSAnimationContext.runAnimationGroup({ context in
    context.duration = 0.4
    backgroundView.animator().alphaValue = isShown ? 1 : 0
}) {
    //
}
```

**Comment by marcelofabri** [comment]
That's a current (known) limitation of this rule and one of the reasons why it's opt-in. SwiftLint currently has no way to know anything about function declarations from other files/modules.

**Comment by notohiro** [comment]
I have same issue when using RxSwift.Observable.subscribe().

**Comment by sindresorhus** [comment]
@marcelofabri For now, couldn't SwiftLint just hard-code to ignore `NSAnimationContext.runAnimationGroup` for this rule?

**Comment by marcelofabri** [comment]
@sindresorhus probably it'd make more sense to create a configuration that can be set with methods to ignore

**Comment by qhhonx** [comment]
> @sindresorhus probably it'd make more sense to create a configuration that can be set with methods to ignore

Any progress for this? We have a similar situation that using `performBatchUpdates` like below will violate *trailing_closure* rule. 
```Swift
collectionView.performBatchUpdates({ // *trailing_closure* rule violation
	// logic
})

collectionView.performBatchUpdates { // compile error
	// logic
}

collectionView.performBatchUpdates({ // works but with given redundant `completion`
	// logic
}, completion: { _ in })
```
Declaration in UIKit:
```Swift
func performBatchUpdates(_ updates: (() -> Void)?, completion: ((Bool) -> Void)? = nil)
```
BTW, we enable *trailing_closure* rule explicitly according to [Google Swift Style Guide](https://google.github.io/swift/#trailing-closures).
